x-pack/legacy/plugins/apm/server/routes/settings.ts

```diff
@@ -20,13 +20,19 @@ import { transactionSampleRateRt } from '../../common/runtime_types/transaction_
 // get list of configurations
 export const agentConfigurationRoute = createRoute(core => ({
   path: '/api/apm/settings/agent-configuration',
-  handler: async req => {
-    await createApmAgentConfigurationIndex(core.http.server);
+  handler: async (req, _, h) => {
+    try {
+      await createApmAgentConfigurationIndex(core.http.server);
 
-    const setup = await setupRequest(req);
-    return await listConfigurations({
-      setup
-    });
+      const setup = await setupRequest(req);
+      return await listConfigurations({ setup });
+    } catch (error) {
+      if (error.statusCode === 403) {
```

Comment:
If we catch 403 authorization errors, how does the user ever see `failurePrompt`? What other status codes should we expect here? just 500?

Reply:
failurePrompt is now make more generic and the statusCode check was removed